src/Types/EnumType.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Types;
+
+use BackedEnum;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\PostgreSQLPlatform;
+use Doctrine\DBAL\Platforms\SQLitePlatform;
+use Doctrine\DBAL\Platforms\SQLServerPlatform;
+use Doctrine\DBAL\Types\Exception\InvalidFormat;
+use Doctrine\DBAL\Types\Exception\InvalidType;
+use Doctrine\DBAL\Types\Exception\ValueNotConvertible;
+use ReflectionClass;
+use Stringable;
+use Throwable;
+use UnitEnum;
+
+use function array_map;
+use function class_exists;
+use function enum_exists;
+use function implode;
+use function in_array;
+use function is_object;
+use function is_string;
+use function sprintf;
+
+final class EnumType extends Type
+{
+    public string $name = 'enum';
```

Comment:
What's the purpose of this property? Types are usually unaware of the name(s) under which they've been registered.